plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -24,22 +24,39 @@ import {
   SingleInstanceGithubCredentialsProvider,
 } from '@backstage/integration';
 import {
+  DeferredEntity,
   EntityProvider,
   EntityProviderConnection,
-  LocationSpec,
   locationSpecToLocationEntity,
 } from '@backstage/plugin-catalog-backend';
 
+import { LocationSpec } from '@backstage/plugin-catalog-common';
+
 import { graphql } from '@octokit/graphql';
 import * as uuid from 'uuid';
+import limiterFactory from 'p-limit';
 import { Logger } from 'winston';
 import {
   readProviderConfigs,
   GithubEntityProviderConfig,
 } from './GithubEntityProviderConfig';
-import { getOrganizationRepositories, RepositoryResponse } from '../lib/github';
+import { getOrganizationRepositories } from '../lib/github';
 import { satisfiesTopicFilter } from '../lib/util';
 
+import { EventParams, EventSubscriber } from '@backstage/plugin-events-node';
+import { PushEvent, Commit } from '@octokit/webhooks-types';
+
+const TOPIC_REPO_PUSH = 'github.push';
+
+type Repository = {
+  name: string;
+  url: string;
+  isArchived: boolean;
+  repositoryTopics: string[];
+  defaultBranchRef: string | null;
```

Comment:
Would it be possible to make this undefined to avoid using null?
```suggestion
  defaultBranchRef?: string;
```

Reply:
Good ideia, I just propagate from Response